### PR TITLE
More info in NS param checking, more selective checking

### PR DESCRIPTION
**Pull request by GiudGiud**
This fixes errors in MSFR with the new conditions on when to check the inlet BCs for scalars

### CODE REVIEW

**Review by grmnptr — DISMISSED** [review]
Very nice change, cleans up a lot of the error messages!

File: modules/navier_stokes/include/actions/NSFVAction.h [hunk 1]
```diff
@@ -316,6 +316,18 @@ class NSFVAction : public Action
   /// Checks that sufficient Rhie Chow coefficients have been defined for the given dimension, used
   /// for scalar or temperature advection by auxiliary variables
   void checkRhieChowFunctorsDefined();
+  /// Check that two parameters are the same size
+  void checkSizeFriendParams(unsigned int param_vector_1_size,
+                             unsigned int param_vector_2_size,
+                             std::string param_name_1,
+                             std::string param_name_2,
+                             std::string object_name_1);
```
Comment (lindsayad) [inline modules/navier_stokes/include/actions/NSFVAction.h:324]:
```suggestion
                             const std::string & param_name_1,
                             const std::string & param_name_2,
                             const std::string & object_name_1);
```

File: modules/navier_stokes/include/actions/NSFVAction.h [hunk 2]
```diff
@@ -316,6 +316,18 @@ class NSFVAction : public Action
...
+  /// Check that a parameter is the the expected size
+  void checkSizeParam(unsigned int param_vector_size,
+                      std::string param_name,
```
Comment (lindsayad) [inline modules/navier_stokes/include/actions/NSFVAction.h:327]:
```suggestion
                      const std::string & param_name,
```

File: modules/navier_stokes/include/actions/NSFVAction.h [hunk 3]
```diff
@@ -316,6 +316,18 @@ class NSFVAction : public Action
...
+                      unsigned int size_wanted,
+                      std::string object_name,
+                      std::string size_source_explanation);
```
Comment (lindsayad) [inline modules/navier_stokes/include/actions/NSFVAction.h:330]:
```suggestion
                      const std::string & object_name,
                      const std::string & size_source_explanation);
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Comment by GiudGiud** [comment]
Should be fixed now

**Review by lindsayad — APPROVED** [review]

**Pull request merged**